portal-impl/src/com/liferay/portal/db/remover/PortalDuplicateRemover.java

```diff
@@ -0,0 +1,212 @@
+package com.liferay.portal.db.remover;
+
+import com.liferay.portal.kernel.dao.jdbc.DataAccess;
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class PortalDuplicateRemover implements DuplicateRemover {
+
+	@Override
+	public Map<Long, List<HashMap<String, String>>> getDuplicates(
+		String tableName, String indexesSQL) {
+
+		Map<String[], String> indexesDuplicatesMap = new LinkedHashMap<>();
+
+		Map<Long, List<HashMap<String, String>>> duplicatesMap =
```

Comment:
I was a bit unsure on using this method to collect all of the entries.  Since there is no exact model to be used,  I applied this method to account for everything.

Reply:
Please see mu comment above